File: tests/android/cryptography/insecure_random_number_generator.java
import java.util.Random;
import java.lang.*;
import java.security.SecureRandom;

// ok: insecure_random_number_generator
SecureRandom sRandom_ok = new SecureRandom();
// ruleid: insecure_random_number_generator
SecureRandom sRandom_ko = new SecureRandom(12);
// ruleid: insecure_random_number_generator
Random random = new Random();
// ruleid: insecure_random_number_generator
int math_r = 1 + Math.random();

File: tests/android/cryptography/insecure_random_number_generator.kt
import java.util.Random
import java.lang.*
import java.security.SecureRandom

// ok: insecure_random_number_generator
val sRandom_ok = SecureRandom();
// ruleid: insecure_random_number_generator
val sRandom_ko = SecureRandom(12);
// ruleid: insecure_random_number_generator
val random = Random();
// ruleid: insecure_random_number_generator
val math_r = 1 + Math.random();